weed/s3api/s3api_object_handlers.go

```diff
@@ -224,3 +227,98 @@ func getObject(vars map[string]string) string {
 	}
 	return object
 }
+
+func (s3a *S3ApiServer) SelectObjectContent(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	bucket := vars["bucket"]
+	object := getObject(vars)
+
+	s3Select, err := s3select.NewS3Select(r.Body)
+	if err != nil {
+		if serr, ok := err.(s3select.SelectError); ok {
+			encodedErrorResponse := encodeResponse(cmd.APIErrorResponse{
+				Code: serr.ErrorCode(),
+				Message: serr.ErrorMessage(),
+				BucketName: bucket,
+				Key: object,
+				Resource: r.URL.Path,
+				RequestID: w.Header().Get(xhttp.AmzRequestID),
+				HostID: "",
+			})
+			writeResponse(w, serr.HTTPStatusCode(), encodedErrorResponse, "application/xml")
+		} else {
+			writeErrorResponse(w, ErrInternalError, r.URL)
+		}
+		return
+	}
+
+	if strings.HasSuffix(r.URL.Path, "/") {
+		writeErrorResponse(w, ErrNotImplemented, r.URL)
+		return
+	}
+
+	destUrl := fmt.Sprintf("http://%s%s/%s%s",
+		s3a.option.Filer, s3a.option.BucketsPath, bucket, object)
+	passThroughResponseSelectObjectContent := func(proxyResonse *http.Response,
+		w http.ResponseWriter) {
+		getObject := func(offset, length int64) (io.ReadCloser, error) {
+			return proxyResonse.Body, nil
+		}
+		if err = s3Select.Open(getObject); err != nil {
+			if serr, ok := err.(s3select.SelectError); ok {
+				encodedErrorResponse := encodeResponse(cmd.APIErrorResponse{
+					Code:       serr.ErrorCode(),
+					Message:    serr.ErrorMessage(),
+					BucketName: bucket,
+					Key:        object,
+					Resource:   r.URL.Path,
+					RequestID:  w.Header().Get(xhttp.AmzRequestID),
+					HostID:     "",
+				})
+				writeResponse(w, serr.HTTPStatusCode(), encodedErrorResponse, mimeXML)
+			} else {
+				writeResponse(w, http.StatusInternalServerError, encodeResponse("not s3select.SelectError"), mimeXML)
+			}
+			return
+		}
+
+		s3Select.Evaluate(w)
+		s3Select.Close()
+	}
+	s3a.proxyToFilerS3Select(w, r, destUrl, passThroughResponseSelectObjectContent)
+}
+
+func (s3a *S3ApiServer) proxyToFilerS3Select(w http.ResponseWriter, r *http.Request, destUrl string,
```

Comment:
Why did you write a new function instead of enhance the existing proxyToFiler func. It seems that only the http METHOD is different.

Reply:
I guess I did it because it's more straight forward. I'll do what you suggested in a future commit. Is that OK? @iliul 
And could you open a new branch "s3-select" to which I can make pull request?